server/client/detectConnect.ts

```diff
@@ -0,0 +1,20 @@
+import { ConnectionSpec } from "../../nets/mod.ts"
+import { Connect, WsConnection } from "../../rpc/mod.ts"
+import { DevnetConnection } from "./DevnetConnection.ts"
+
+export function detectConnect(
+  spec: ConnectionSpec,
+  targets: Record<string, ConnectionSpec>,
+): Connect {
+  const targetName = Deno.env.get("CAPI_TARGET")
+  if (targetName) {
+    const target = targets[targetName]
+    if (target) spec = target
+  }
+  switch (spec.type) {
+    case "WsConnection":
+      return WsConnection.bind(spec.discovery)
+    case "DevnetConnection":
+      return DevnetConnection.bind(spec.discovery)
+  }
+}
```

Comment:
Not a fan of parameter reassignment so something like this could work better.

```suggestion
export function detectConnect(
  spec: ConnectionSpec,
  targets: Record<string, ConnectionSpec>,
): Connect {
  const targetName = Deno.env.get("CAPI_TARGET") ?? ""
  return bindConnection(targets[targetName] ?? spec)
}

function bindConnection(spec: ConnectionSpec) {
  switch (spec.type) {
    case "WsConnection":
      return WsConnection.bind(spec.discovery)
    case "DevnetConnection":
      return DevnetConnection.bind(spec.discovery)
  }
}
```